### PR TITLE
sample that shows how we can implement additional authorization requirements with an external authorization service

### PR DESCRIPTION
**Pull request by seekdavidlee**
***EDIT by @Rick-Anderson :*** 
contributes to [#27317](https://github.com/dotnet/AspNetCore.Docs/issues/27317)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs.Samples?pullRequest=104) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs.Samples?pullRequest=104) before we can accept your contribution.<br/><br/>:x: seekdavidlee [sign now](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs.Samples?pullRequest=104)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/AspNetCore.Docs.Samples?pullRequest=104) it.</sub>

**Comment by Rick-Anderson** [comment]
@seekdavidlee please sign the [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs.Samples?pullRequest=104) before we can accept your contribution.

File: samples/aspnetcore-authz-with-ext-authz-service/.gitignore [hunk 1]
```diff
@@ -0,0 +1,352 @@
+## Ignore Visual Studio temporary files, build results, and
```
Comment (Rick-Anderson) [inline samples/aspnetcore-authz-with-ext-authz-service/.gitignore:1]: Delete

File: samples/aspnetcore-authz-with-ext-authz-service/Contoso.API/Program.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
+using Contoso.API.Authorization;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.Identity.Web;
+using Microsoft.Identity.Web.Resource;
+
+namespace Contoso.API
+{
+    public class Program
```
Comment (Rick-Anderson) [inline samples/aspnetcore-authz-with-ext-authz-service/Contoso.API/Program.cs:10]: Use template generated Program.cs

File: samples/aspnetcore-authz-with-ext-authz-service/Contoso.API/Program.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        public static void Main(string[] args)
+        {
+            var builder = WebApplication.CreateBuilder(args);
+
+            builder.Services.AddHttpClient(Constants.SecurityAPIClient, httpClient =>
+            {
+                httpClient.BaseAddress = new Uri("https://localhost:7123/");
+            });
+
+            // Add services to the container.
```
Comment (Rick-Anderson) [inline samples/aspnetcore-authz-with-ext-authz-service/Contoso.API/Program.cs:21]: remove template generated comments

**Comment by Rick-Anderson** [comment]
@seekdavidlee 
image

**Comment by seekdavidlee** [comment]
> @seekdavidlee image

This is completed.

File: samples/aspnetcore-authz-with-ext-authz-service/Contoso.API/Program.cs (hunk 2 above)
Comment (Rick-Anderson) [reply samples/aspnetcore-authz-with-ext-authz-service/Contoso.API/Program.cs:10]: > @Rick-Anderson - This is template generated. Can you explain?

Create an app with the templates, that is with Visual Studio or `dotnet new`. We should always start with the tool generated code.

Comment (seekdavidlee) [reply samples/aspnetcore-authz-with-ext-authz-service/Contoso.API/Program.cs]: Yes, this solution/project is generated with Visual Studio. I went ahead and tested with dotnet new, and it seemed dotnet new produces a version with minimal api.

File: samples/aspnetcore-authz-with-ext-authz-service/Contoso.API/Authorization/CanGetWeatherAuthorizationHandler.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
+using Microsoft.AspNetCore.Authorization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Contoso.API.Authorization
```
Comment (Rick-Anderson) [inline samples/aspnetcore-authz-with-ext-authz-service/Contoso.API/Authorization/CanGetWeatherAuthorizationHandler.cs:5]: Use [File Scoped Namespaces](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces)

Comment (seekdavidlee) [reply samples/aspnetcore-authz-with-ext-authz-service/Contoso.API/Authorization/CanGetWeatherAuthorizationHandler.cs]: updated, I did not know about this feature, appreciate the link!

**Comment by Rick-Anderson** [comment]
@adityamandaleeka please assign a reviewer

**Comment by adityamandaleeka** [comment]
@HaoK Can you please review this when you have some time?

**Comment by seekdavidlee** [comment]
> @HaoK Can you please review this when you have some time?

Hi @HaoK @adityamandaleeka - Wanted to ping again to see if we have an update. Thanks!

File: samples/aspnetcore-authz-with-ext-authz-service/Contoso.API/Controllers/WeatherForecastController.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+[Authorize(Policy = AppConstants.CanGetWeatherPolicyName)]
+[ApiController]
+[Route("[controller]")]
+public class WeatherForecastController : ControllerBase
+{
+    private static readonly string[] Summaries = new[]
+    {
+    "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
+};
```
Comment (HaoK) [inline samples/aspnetcore-authz-with-ext-authz-service/Contoso.API/Controllers/WeatherForecastController.cs:14]: nit indent

**Review by HaoK — APPROVED** [review]
Sample looks good, sorry for the delay

**Pull request merged**